activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -64,7 +64,7 @@ def visit_PrimaryKeyDefinition(o)
           end
 
           def visit_ForeignKeyDefinition(o)
-            sql = <<-SQL.strip_heredoc
+            sql = <<-SQL.strip_heredoc.dup
```

Comment:
Is this `dup` necessary? Looks like that `strip_heredoc` returns new string and we have no `strip_heredoc.dup`.
https://github.com/rails/rails/blob/v5.1.2/activesupport/lib/active_support/core_ext/string/strip.rb#L21

```
% git grep 'strip_heredoc\.dup' | wc -l
0
```